### PR TITLE
Remove unnecessary fields in the sub-event array of super events

### LINKED ISSUE #836: [BUG] Some fields in the sub events array in super events should be deleted before publishing
**Describe the bug**
There are some extra fields like `name` and `eventid` in the sub-event array of super events.

**To Reproduce**
1. Create a new pending super event
2. Create a new pending event
3. Add the new event to the super event as a sub-event
4. Publish the pending super event
5. Unnecessary fields show up in the build block DB

**Expected behavior**
Only `id` `isFeatured` and `track` should be in the DB

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version 98.0.4758.82

### PR DESCRIPTION
**Pull request by clav1d**
## Description
Remove all keys in the sub-events array except for those in the schema.

**Fixes #836 **

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by sandeep-ps** [comment]
@clav1d, Please remove this PR from the "Platform Development" GitHub project as this is already linked to the issue and the issue is in the same project. This is to avoid duplicate entries. Thanks.

**Comment by clav1d** [comment]
> @clav1d, Please remove this PR from the "Platform Development" GitHub project as this is already linked to the issue and the issue is in the same project. This is to avoid duplicate entries. Thanks.

So only the issue needs to be put in the project?

**Comment by sandeep-ps** [comment]
Yes, that's right. Thanks.

**Comment by sandeep-ps** [comment]
I'm still seeing unnecessary fields in the published super event (see below). Could you please look into this? 

image

**Comment by clav1d** [comment]
> I'm still seeing unnecessary fields in the published super event (see below). Could you please look into this?
> 
> image

Sure, I'll look into this. Are these fields still showing up on the Events Manager's database or the Events Building Block database?

**Comment by sandeep-ps** [comment]
> > I'm still seeing unnecessary fields in the published super event (see below). Could you please look into this?
> > image
> 
> Sure, I'll look into this. Are these fields still showing up on the Events Manager's database or the Events Building Block database?

This is the GET /events/<id> response from the Events Building Block after publishing the super event. I followed the steps to repeat in the bug report but added two sub-events instead of one. Thanks.

**Comment by clav1d** [comment]
> > > I'm still seeing unnecessary fields in the published super event (see below). Could you please look into this?
> > > image
> > 
> > 
> > Sure, I'll look into this. Are these fields still showing up on the Events Manager's database or the Events Building Block database?
> 
> This is the `GET /events/<id>` response from the Events Building Block after publishing the super event. I followed the steps to repeat in the bug report but added two sub-events instead of one. Thanks.

Thanks for the information. I'll be working on fixing this.

**Comment by clav1d** [comment]
Hi @sandeep-ps , I believe this bug is fixed now. Please review again when you are available.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix the issue that publish the super event without sub events. [#827](https://github.com/rokwire/events-manager/issues/827)
 - Update superevent when its subevent has been removed. [#838](https://github.com/rokwire/events-manager/issues/838)
 - Fix subevent publish status on its super event detail page. [#840](https://github.com/rokwire/events-manager/issues/840)
+- Remove unnecessary fields in sub-event array of super events. [#836](https://github.com/rokwire/events-manager/issues/836)
```
Comment (sandeep-ps) [inline CHANGELOG.md:42]: Please move this to the Unreleased section.

**Comment by sandeep-ps** [comment]
Please fix the merge conflicts. Thanks.

**Comment by sandeep-ps** [comment]
I think I will fix the merge conflicts and merge this PR.

**Review by sandeep-ps — APPROVED** [review]

**Pull request merged**